### ISSUE TITLE
Parse error when using rest parameter together with destructuring for function arguments

### ISSUE THREAD

**Issue by n0v1** [SonarSource/SonarJS#1125]
**I want to report a bug.**

**SonarJS version:** 5.0 (build 6962)
**SonarQube version:** 7.4.0.18908

SonarJS fails to parse the following code that uses a rest parameter together with array/object destucturing for function arguments.

**Reproducer**
```javascript
const stampit = require('stampit')

function test (...[, {instance: thisInstance}]) {
  thisInstance.foo = 'bar'
  return thisInstance
}
const MyStamp = stampit.init(test)
const myStamp = MyStamp()
console.log(myStamp.foo)
```

**Logs**
```
ERROR: Unable to parse file: file:///test.js
ERROR: Parse error at line 3 column 47:

 1: const stampit = require('stampit')
 2:
 3: function test (...[, {instance: thisInstance}]) {
                                                  ^
 4:   thisInstance.foo = 'bar'
 5:   return thisInstance
 6: }
 7: const MyStamp = stampit.init(test)
 8: const myStamp = MyStamp()
 9: console.log(myStamp.foo)
10:

INFO: Sensor SonarJS [javascript] (done) | time=67ms
```

**Expected behavior**

Should parse without problems since this is valid ES6.

**Comment by vilchik-elena** [comment]
Hello @n0v1 

Currently we are gradually migrating SonarJS from our own parser to esprima-compatible (first step is done in SonarJS 5.0). I see your concern, but this is exactly why we want to use another parser, it's too painful to support all favours and syntaxes of JS.

So even if there is parsing error coming from our parser, hopefully another parser processes this correctly and some part of rules is executed successfully.

So I apology for the inconvenience you have and hope that SonarJS will be migrated fully to the third-party parser pretty soon.

**Comment by n0v1** [comment]
@vilchik-elena thanks for your response. I agree that using an established parser for this projec is a good idea. Is there a ticket to follow the progress of this task?

**Comment by vilchik-elena** [comment]
@n0v1 see it here https://jira.sonarsource.com/browse/MMF-1409
